contracts/external/cover/CoverBroker.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-v4/access/Ownable.sol";
+
+import "../../interfaces/ICover.sol";
+import "../../interfaces/ICoverBroker.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/INXMToken.sol";
+
+
+/// @title Cover Broker Contract
+/// @notice Enables non-members of the mutual to purchase cover policies.
+/// Supports ETH, NXM, and ERC20 asset payments which are supported by the pool.
+/// @dev See supported ERC20 payment methods via pool.getAssets.
+contract CoverBroker is ICoverBroker, Ownable {
+  using SafeERC20 for IERC20;
+
+  // Immutables
+  ICover public immutable cover;
+  IMemberRoles public immutable memberRoles;
+  IPool public immutable pool;
```

Comment:
The pool contract should not be immutable as it can change letting this contract call into the old pool.

Reply:
We'll have to add an immutable master address instead and pull the pool contract address from it.

Reply:
good catch